mindsdb/integrations/postgres_handler/postgres_handler.py

```diff
@@ -0,0 +1,101 @@
+from mindsdb.integrations.libs.base_handler import DatabaseHandler
+import psycopg
+from mindsdb.utilities.log import log
+from contextlib import closing
+from mindsdb_sql import parse_sql
+
+class PostgresHandler(DatabaseHandler):
+    """
+    This handler handles connection and execution of the PostgreSQL statements. 
+    """
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name)
+        self.parser = parse_sql
+        self.dialect = 'postgresql'
+        self.database = kwargs.get('database')
+        self.host = kwargs.get('host')
+        self.port = kwargs.get('port')
+        self.user = kwargs.get('user')
+        self.password = kwargs.get('password')
+
+    def __connect(self):
+        """
+        Handles the connection to a PostgreSQL database insance.
+        """
+        #TODO: Check psycopg_pool
+        connection = psycopg.connect(f'host={self.host} port={self.port} dbname={self.database} user={self.user} password={self.password}', connect_timeout=10)
+        return connection
+
+    def check_status(self):
+        """
+        Check the connection of the PostgreSQL database
+        """
+        connected = False
+        try:   
+            con = self.__connect()
+            with closing(con) as con:
+                with con.cursor() as cur:
+                    cur.execute('select 1;')
+            connected = True
+        except Exception as e:
+            log.error(f'Error connecting to PostgreSQL {self.database} on {self.host}!')
+            pass
+        return connected
```

Comment:
will be good return here status and reason. So we will be able show to user why connection is not working, for example:
```
 {
     'success': True
 }
 {
     'success': False,
     'error': 'password is wrong'
 } 
```